pkg/chain/confirmation.go

```diff
@@ -0,0 +1,32 @@
+package eth
+
+import (
+	"fmt"
+)
+
+// WaitForChainConfirmation ensures that after receiving specific number of block
+// confirmations the state of the chain is actually as expected. It waits for
+// predefined number of blocks since the start block number provided. After the
+// required block number is reached it performs a check of the chain state with
+// a provided function returning a boolean value.
+func WaitForChainConfirmation(
```

Comment:
Can we put this guy in `keep-common`? We already have `MiningWaiter` there and at some point, we'll want to integrate this check with `keep-core` as well.

Reply:
Can be a little bit problematic as it depends on `chain.Handle`. From the other side, passing the block counter alone seems a bit weird.

Reply:
Does it? I think it is fine - this function does not need anything else from the chain than the acces to the util counting blocks. We'll pass exactly what this function needs.

What feels a bit weird to me is this function in this package next to interface declarations, I think it's a perfect candidate for `keep-common` chain utilities. Also, what triggers a warning to me is `logger` declaration we put in `chain.go` next to interface declarations. Logger should live next to the implementation, not the interface. 

Yeah, let's please make it a `keep-common` code.

Reply:
Ok, done.